EXT_BeamPhysics.md

```diff
@@ -0,0 +1,264 @@
+Extension to the openPMD Standard for Describing Particle Beams and X-rays
+==========================================================================
+
+Version 2.0.0
+
+Overview
+--------
+
+The `BeamPhysics` extension to the openPMD standard is meant for describing particles and fields commonly encountered
+in accelerator physics simulations.
+
+How to Use this Extension
+-------------------------
+
+The `BeamPhysics` extension to the openPMD standard is indicated in a data file by the setting of the `openPMDextension` attribute:
+```
+  openPMDextension = BeamPhysics;SpeciesType
+```
+
+Note: The `speciesType` extension must be used when using the `BeamPhysics` extension.
+
+Definitions
+-----------
+
+- **Lattice**: A **lattice** is the arrangement of elements in a machine such as a particle accelerator.
+
+- **Global coordinate system**: The **global** coordinate system is a coordinate system that is used to describe the position and orientation of machine elements in space. That is, the **global** coordinate system is fixed with respect to the building or room where the machine is placed independent of the machine itself.
+
+- **Lattice coordinate system**: The curvilinear coordinate system whose "longitudinal" coordinate (**s**) typically runs through the nominal centers of the elements in the machine. The **lattice** coordinate system is often used to describe particle positions.
+
+- **Macro-particle**: Macro-particles are simulation particles that represent multiple real particles. The number of real particles that a macro-particle represents affects the calculation of the field due to a macro-particle but does not affect tracking.
+
+- **Polar coordinates**: **Polar** coordinates are **(r, theta, phi)** where **r** is the radius, **theta** is the angle from the **z** or **s** axis, and **phi** is the projected azimuthal angle in the **(x, y)** plane.
+
+- **Particle Root Group**: The **Particle Root Group** is a group for specifying a group of particles. See the Base Standard for more information.
+
+- **Phase Space Coordinates**: Phase space coordinates are ordered (x, px, y, py, z, pz).
+
+Notes:
+------
+
+- When using the **lattice** coordinate system, the `position` coordinates are **(x, y, s)** or **(x, y, z)** where, nominally, **x** is the "horizontal" component, **y** is the "vertical" coordinate, and **s** or **z** is the lattice longitudinal coordinate.
+
+Additional File Root Group (path `/`) Attributes
+------------------------------------------------
+
+The following attributes are defined for the file root group.
+
+- `latticeName`
+  - Type: Optional *(string)*
+  - Description: The name of the lattice.
+
+- `latticeFile`
+  - Type: Optional *(string)*
+  - Description: The location of the root lattice file.
+
+
+`Particle Root Group` Attributes
+--------------------------------
+
+For each **particle root group** the following attributes are defined:
+
+- `charge`
```

Comment:
VC: let us make this a *scalar record* and use a *constant* scalar record if it's the same for all particles of a species.

Reply:
OK.